security/src/main/java/uk/ac/ox/ndph/mts/security/authorisation/AuthorisationService.java

```diff
@@ -162,85 +141,33 @@ public boolean authoriseUserRoles(String userIdentityParam) {
-
-        try {
-            Objects.requireNonNull(sitesReturnObject, "sites can not be null");
-
-            List<SiteDTO> sites = sitesReturnObject.stream()
-                    .map(siteObject -> {
-                        var siteId =  siteUtil.getSiteIdFromObj(siteObject, "getSiteId");
-                        var parentSiteId =  siteUtil.getSiteIdFromObj(siteObject, "getParentSiteId");
-                        return new SiteDTO(siteId, parentSiteId);
-                    }).collect(Collectors.toList());
-
-            String userId = securityContextUtil.getUserId();
-            String token = securityContextUtil.getToken();
-            Consumer<org.springframework.http.HttpHeaders> authHeaders = PractitionerServiceClient.bearerAuth(token);
-
-            List<RoleAssignmentDTO> roleAssignments =
-                new ArrayList<>(practitionerServiceClient.getUserRoleAssignments(userId, authHeaders));
-
-            if (role != null) {
-                roleAssignments.removeIf(ra -> !ra.getRoleId().equalsIgnoreCase(role));
-            }
-
-            Set<String> userSites = siteUtil.getUserSites(sites, roleAssignments);
-
-            sitesReturnObject.removeIf(siteObject ->
-                    !userSites.contains(siteUtil.getSiteIdFromObj(siteObject, "getSiteId")));
-
-            if (sitesReturnObject.isEmpty()) {
-                return false;
-            }
-
-            return true;
-        } catch (Exception e) {
-            return false;
-        }
-
-    }
-
     /**
      * Validate if the role assignments have the required permission linked to them.
-     * @param requiredPermission action required permission
-     * @param roleAssignments user role assignments
+     * @param reqPerm action required permission
+     * @param roleAssmnts user role assignments
      * @return true if required permission is present in one of the roles
      */
-    private List<RoleAssignmentDTO> getRolesAssignmentsWithPermission(String requiredPermission,
-                                                                      List<RoleAssignmentDTO> roleAssignments) {
-
-        List<String> roleIds = roleAssignments.stream()
-                .map(RoleAssignmentDTO::getRoleId).collect(Collectors.toList());
-
+    private List<RoleAssignmentDTO> getRoleAssmntsWtPerm(String reqPerm,
+                                                         List<RoleAssignmentDTO> roleAssmnts) {
         //get permissions for the the practitioner role assignments
         //and filter role assignments to be only those which have the required permission in them
-        Set<String> rolesWithPermission = roleServiceClient.getRolesByIds(roleIds,
-                RoleServiceClient.bearerAuth(securityContextUtil.getToken())).stream()
-                .filter(roleDto -> hasRequiredPermissionInRole(roleDto, requiredPermission))
+        Page<RoleDTO> roleDTOs = roleServClnt.getPage(0, 500, RoleServiceClient.bearerAuth(getToken()));
```

Comment:
If this is paged then need to handle paging to get all the roles, don't assume there is some limit.

Reply:
@darrengsmithson put a TODO here ? wdyt